runtime/lua/vim/_editor.lua

```diff
@@ -997,6 +1000,10 @@ function vim._expand_pat(pat, env)
     insert_keys(vim._extra)
   end
 
+  if final_env == vim.lsp or final_env == vim.treesitter or final_env == vim.provider then
```

Comment:
does this work (and avoids future maintenance)?
```suggestion
  elseif final_env._submodules then
```